cranelift/codegen/src/verifier/mod.rs

```diff
@@ -656,12 +656,54 @@ impl<'a> Verifier<'a> {
                     ));
                 }
             }
-            LoadNoOffset {
-                opcode: Opcode::Bitcast,
-                flags,
-                arg,
+            LoadNoOffset { opcode, flags, arg } => {
+                if opcode == Opcode::Bitcast {
```

Comment:
Let's keep the original nested match here -- it's cleaner than ad-hoc ifs inside the match arm. Something like

```
LoadNoOffset { opcode: Opcode::Bitcast, flags, arg } => { /* original */ }
LoadNoOffset { opcode, flags, arg } if opcode.can_load() => { /* verify address */ }
```

and likewise below: pull the ifs that guard with `can_load()`/`can_store()` into match guards.

Reply:
I really like this feedback, this is something I'll adopt in my own code!